### PR TITLE
[bugfix] Orca sparkxshards uncache bugfix

### PR DESCRIPTION
**Pull request by cyita**
related issue: [XShards __del__ gets exception if xshards was not cached before #2433](https://github.com/intel-analytics/analytics-zoo/issues/2433)

### CODE REVIEW

File: pyzoo/zoo/orca/data/shard.py [hunk 1]
```diff
@@ -145,7 +145,8 @@ def cache(self):
 
     def uncache(self):
         self.user_cached = False
-        self.rdd.unpersist()
+        if self.is_cached():
+            self.rdd.unpersist()
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/shard.py:149]: try ... catch ...?

**Comment by jason-dai** [comment]
LGTM

**Comment by jenniew** [comment]
Seems self.user_cached is not used anywhere. We just set self.user_cached. When we want to check if cached, we call is_cached() which is to check self.rdd.is_cached.

**Comment by jason-dai** [comment]
> Seems self.user_cached is not used anywhere. We just set self.user_cached. When we want to check if cached, we call is_cached() which is to check self.rdd.is_cached.

it's used in the `_uncache` method

**Comment by jenniew** [comment]
Sorry I'm a little confusing. I think we don't need self.user_cached flag.  We create cache() and uncache()/_uncache() function. cache() would wrap self.rdd.cache(). uncache()/_uncache() is to check self.rdd.is_cache, then unpersist(). When construct xshards, we call cache() to cache. When need to uncache, call uncache()/_uncache(). Why we need to tell user_cached or not user_cached.

**Comment by cyita** [comment]
> Sorry I'm a little confusing. I think we don't need self.user_cached flag. We create cache() and uncache()/_uncache() function. cache() would wrap self.rdd.cache(). uncache()/_uncache() is to check self.rdd.is_cache, then unpersist(). When construct xshards, we call cache() to cache. When need to uncache, call uncache()/_uncache(). Why we need to tell user_cached or not user_cached.

If the user calls `cache()` on a SparkXShard, this shard will not be automatically uncached after `transform_shard`.

**Pull request merged**